report/5.Solution/DockerBuild.tex

```diff
@@ -0,0 +1,107 @@
+This complicated things a bit, as none of the group members were interested in installing such an old operating system on their personal computer, and instead Docker was used to avoid this.
+
+\subsection{Docker}\label{subsec:docker}
+Docker is a way to deploy a consistent environment for applications to run in, also known as containerization\cite{dockerdocstart}.
+A Docker deployment consists of two different concepts, the container and an image.
+\begin{description}
+    \item [Images] are executables that includes all the required information to instantiate containers.
+    \item[Containers] are the runtime instances of an image, the running process of an image.
+\end{description}
+
+A Docker container is in broad strokes the same as a virtual machine, although a Docker container only uses the required resources, while a virtual machine is allowed access to more resources than is typically needed.
+Docker runs natively on Linux, and takes no more memory than other executables.
+
+A Docker deployment consists of two different concepts, the container and an image.
+\begin{description}
+    \item[Images] are executables that includes all the required information to instantiate containers.
+    \item[Containers] are the runtime instances of an image, the running process of an image.
+\end{description}
+The concept of containerization as a way to ensure consistency in deployment of software, was perfect for the purposes of the group.
+Docker is supported on all 3 major platform, and while both Docker for Windows and OSX run in a virtual machine, this is a non-issue.
+
+A potential issue was nxtOSEK not being officially supported past Ubuntu 11.10, and Docker officially supporting Ubuntu 18.04, 16.04 and 14.04 /cite{dockerubuntu}.
+However, a quick test proved that nxtOSEK still worked fine on Ubuntu 14.04.
+
+\subsection{Docker implementation}\label{subsec:dockerimplementation}
+The actual Docker build process is somewhat obscure.
+
+The first step is relatively straightforward with \textit{FROM ubuntu:14.04} simply specifying which version of Ubuntu to use.
+The next bit simply installs the packages required by nxtOSEK.
+\begin{lstlisting}[language=docker,label={lst:dockerimplementation1},caption={Version definition and installation of packages required by nxtOSEK}]
+    FROM ubuntu:14.04
+
+    # Install required packages listed by nxtOSEK
+    RUN apt-get update
+    RUN apt-get -y install tk-dev ncurses-dev libmpfr-dev wget gzip tar software-properties-common xvfb
+\end{lstlisting} 
+
+The next part is the most obscure part, as wine is installed to be able to run Windows programs.
+This is done as the tool to build OIL files, only exists for Windows.
+NeXT Tools is an utility program that includes various utilities for use with nxtOSEK, including an enhanced OIL file compile\cite{nxttool}.
+It also includes an additional nxtOSEK requirement, texinfo, which is the official GNU documentation format\cite{texinfo}.
```

Comment:
Maybe add that this could not be fetched from apt repo, since that included a newer version that was not compatible with the nxtOSEK required (older) version. Hence the downgrading comment from before.

Reply:
Well, actually kind of explained by the comment "
    # Download a specific version of texinfo, the one in 14.04 
    # sources is not compatible with the expected version
"
But I still think it is worth being explicit about